devel/onetbb/files/patch-onetbb-older-platforms.diff

```diff
@@ -0,0 +1,513 @@
+-    TBB_EXPORT void __TBB_EXPORTED_FUNC notify_bounded_queue_monitor( concurrent_monitor* monitors, std::size_t monitor_tag
+-                                                            , std::size_t ticket );
++    TBB_EXPORT void __TBB_EXPORTED_FUNC notify_bounded_queue_monitor( concurrent_monitor* monitors, std::size_t monitor_tag,
++                                                            std::size_t ticket );
+     TBB_EXPORT void __TBB_EXPORTED_FUNC wait_bounded_queue_monitor( concurrent_monitor* monitors, std::size_t monitor_tag,
+                                                             std::ptrdiff_t target, d1::delegate_base& predicate );
+ } // namespace r1
+diff --git a/include/oneapi/tbb/task_group.h b/include/oneapi/tbb/task_group.h
+index 2bbacd55..8981b54e 100644
+--- a/include/oneapi/tbb/task_group.h
++++ b/include/oneapi/tbb/task_group.h
+@@ -1,5 +1,5 @@
+ /*
+-    Copyright (c) 2005-2022 Intel Corporation
++    Copyright (c) 2005-2023 Intel Corporation
+ 
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+@@ -187,6 +187,23 @@ private:
+     };
+     task_group_context_version my_version;
+ 
++  #if defined __APPLE__ && defined __ppc__
++    /** bool is 32 bits on Darwin ppc32 (but not on ppc64): https://gcc.gnu.org/bugzilla/show_bug.cgi?id=107590
++    https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/LowLevelABI/100-32-bit_PowerPC_Function_Calling_Conventions/32bitPowerPC.html. */
++    //! The context traits.
++    struct context_traits {
++        bool fp_settings        : 4;
++        bool concurrent_wait    : 4;
++        bool bound              : 4;
++        bool reserved1          : 4;
++        bool reserved2          : 4;
++        bool reserved3          : 4;
++        bool reserved4          : 4;
++        bool reserved5          : 4;
++    } my_traits;
++
++    static_assert(sizeof(context_traits) == 4, "Traits shall fit into 4 bytes.");
++  #else // Everything else besides Darwin PPC:
+     //! The context traits.
+     struct context_traits {
+         bool fp_settings        : 1;
+@@ -200,6 +217,7 @@ private:
+     } my_traits;
+ 
+     static_assert(sizeof(context_traits) == 1, "Traits shall fit into one byte.");
++  #endif
+ 
+     static constexpr std::uint8_t may_have_children = 1;
+     //! The context internal state (currently only may_have_children).
+@@ -417,7 +435,11 @@ private:
+     friend class task_group_base;
+ }; // class task_group_context
+ 
+-static_assert(sizeof(task_group_context) == 128, "Wrong size of task_group_context");
++#if defined __APPLE__ && defined __POWERPC__
++    // static_assert(sizeof(task_group_context) == 128, "Wrong size of task_group_context");
++#else
++    static_assert(sizeof(task_group_context) == 128, "Wrong size of task_group_context");
```

Comment:
Negate the condition and wrap this line, like is done in `conformance_allocators.cpp`.